### DISCUSSION TITLE
When running unit tests for dropped events; how can I disable the stderr message "ERROR transform"?

### DISCUSSION THREAD

**Discussion by patrick-oday-cruise** [vectordotdev/vector#17312] [Q&A] [answered]
I have a unit test that uses a log event that is supposed to:

1. Test a remap transform that specifies `reroute_dropped = true` and `drop_on_abort = true`
2. Trigger an abort via `parse_linux_authorization!`
3. Then capture the dropped event by a downstream dropped handler.

The test works as expected except that it prints out to stderr the details of the dropped event. Is there a way to configure the test so that it's aware that the transform will drop the event and not print to stderr during testing?

The stderr that I'm referring to:

> 2023-05-04T22:50:10.449577Z ERROR transform{component_kind="transform" component_id=syslog_transform component_type=remap component_name=syslog_transform}: vector::internal_events::remap: Mapping failed with event. error="function call error for \"parse_linux_authorization\" at (22:58): unable to parse input as valid syslog message" error_type="conversion_failed" stage="processing" internal_log_rate_limit=true
> 2023-05-04T22:50:10.449702Z ERROR transform{component_kind="transform" component_id=syslog_transform component_type=remap component_name=syslog_transform}: vector_common::internal_event::component_events_dropped: Events dropped intentional=false count=1 reason="Mapping failed with event." internal_log_rate_limit=true

I want to have tests that ensure the messages that should be dropped are correctly dropped. The behavior of parse_key_value wasn't exactly as I had expected but I was unaware until testing the failure states.

**Comment by spencergilbert** [comment]
I don't believe that's an existing feature - have you tried redirecting STDOUT vs STDERR?

**Reply by patrick-oday-cruise** [reply]
I have tried redirecting stderr to hide the in-progress intentional errors but I'm concerned that it may also hide other important information, especially when someone is iterating on a test. It's that I'd want stderr messages when a test that is supposed to drop a log event doesn't and stderr messages when a test that isn't supposed to drop a log event does.

**Reply by spencergilbert** [reply, marked as answer]
Got it, definitely not supported today - please open a feature request issue with the use case and details you provided. Thanks!
